pygmt/clib/session.py

```diff
@@ -1506,9 +1506,9 @@ def virtualfile_from_vectors(
                 strings = string_arrays[0]
             elif len(string_arrays) > 1:
                 strings = np.array(
-                    [" ".join(vals) for vals in zip(*string_arrays, strict=True)]
+                    [" ".join(vals) for vals in zip(*string_arrays, strict=True)],
+                    dtype=np.str_,
```

Comment:
Specifying `dtype` is not necesary here, but I feel it's good to expicitly tell that here we're expecting a np.str_ array.